bundler/spec/bundler/settings_spec.rb

```diff
@@ -351,4 +351,30 @@
       expect(settings["mirror.https://rubygems.org/"]).to eq("http://example-mirror.rubygems.org")
     end
   end
+
+  describe "default_cli_command validation" do
+    let(:settings) { described_class.new }
```

Comment:
I think this may be the issue making the specs leak. You're overriding the global `subject(:settings)` which correctly uses `bundled_app` as the settings root.